Fix/load search results only once

ulferts opened the pull request:
Reduces the times, search results are loaded from three to one and by that improves performance quite a bit.

https://community.openproject.com/projects/openproject/work_packages/29715

ulferts committed:
avoid loading query initially again when initializing the query props

ulferts committed:
skip first subscription caused by initialization

And by that skip loading the query again

ulferts commented:
The tests broken are unit tests which I cannot have caused as I only changed front end code.

wielinde commented on frontend/src/app/modules/global_search/global-search-work-packages.component.ts:
```diff
@@ -108,13 +100,13 @@ export class GlobalSearchWorkPackagesComponent extends WorkPackageEmbeddedTableC
     )
     .subscribe(([newSearchTerm, newProjectScope]) => {
       this.WpFilter.visible = false;
-      this.firstTimeLoadForm = true;
```
@ulferts If we remove this line, we run into trouble. Say, you search, open "Advanced Filters" then the searched term shall be set as value of the SearchFilter. Then you trigger a new search through the GlobalSearchInput and open "Advanced filters" again. If we remove this line, the value of the SearchFilter does not update, and that is bad. Obviously, loading the form everytime we perform a new search isn't optimal either but at least "it works".

ulferts replied:
Will take a look at it, @wielinde. Thanks for finding.

wielinde reviewed (changes requested):
@ulferts please see my comment/discussion above.

ulferts commented:
Pushed a new version where I removed the `firstTimeLoadForm` variable altogether. As  `loadQuery` is now only called in cases where the form is also required to be loaded, removing should be save.

wielinde commented:
Hey @ulferts, on a first check I found that the search filter value in the advanced filters does not updated when you fire a new search from within the global search input.

ulferts commented:
@wielinde will take a look

ulferts committed:
only load from work_packages api on filter changes

The query will not change in the sense that there will be changes in the available filters or allowed filter values

ulferts committed:
extract method and fix indentation

wielinde commented:
@ulferts thank you!
